src/MasonryFlashList.tsx

```diff
@@ -179,6 +179,10 @@ const MasonryFlashListComponent = React.forwardRef(
                     ...innerArgs,
                     item: innerArgs.item.originalItem,
                     index: innerArgs.item.originalIndex,
+                    extraData: {
```

Comment:
Why do you need the column number? Columns are abstracted because if we build a way in the future to change column span then `columnIndex` will need to be deprecated. If an item spans across columns then API should be `startColumn` and `endColumn`.
Also, changing extraData ref can cause unwanted rendering as we guarantee forwarding the same reference to components but this is going to change that.